Update to new ITC API related to #228 (application.platform/version/versionSets)

lacostej opened the pull request:
PR is ready for review. I did 3 main things:
* updated unit test data to new format
* refactored the `client.app_version` method to allow querying less the server when we already have the data
* replaced the `application.platform` field with a method that tries to find (reused from original `client.app-version` method

I only tested with unit tests so far.

TODO:
* ~~I probably want to rework the `live_version` change and not remove so much. We lack unit test to cover that case~~
* ~~I can squash all that into one commit if we agree we want to keep the refactoring to reduce server calls~~.

KrauseFx commented:
:+1:

KrauseFx commented:
> I only tested with unit tests so far.

Is this still current, or did you actually test it?

lacostej commented:
@KrauseFx I am working on another PR to clean some data and add some coverage for a case that I think I broke in this one. I'll submit the other one first, then update #245

lacostej commented:
@KrauseFx I added the missing test (e3ea90a), rebased and verified that things work as they should under unit testing. I also used the opportunity to make some additional cleanups (828e555)

KrauseFx commented:
Looks good to me. Did you test this again the actual iTunes Connect back-end service?

lacostej commented:
Now tested against ITC.

KrauseFx commented:
Looks great, thank you @lacostej :+1:
